src/Corax/Indexing/IndexWriter.cs

```diff
@@ -84,12 +82,12 @@ public void UpdateDynamicFieldsMapping(IndexFieldsMapping current)
         // to explicitly provide the index writer with opening semantics and also every new
         // writer becomes essentially a unit of work which makes reusing assets tracking more explicit.
 
-        private IndexWriter(IndexFieldsMapping fieldsMapping, bool phraseQuerySupport)
+        private IndexWriter(IndexFieldsMapping fieldsMapping, SupportedFeatures supportedFeatures)
         {
             _indexDebugDumper = new IndexOperationsDumper(fieldsMapping);
             _builder = new IndexEntryBuilder(this);
             _fieldsMapping = fieldsMapping;
-            _phraseQuerySupport = phraseQuerySupport;
+            _supportedFeatures = supportedFeatures ?? new(); // if not explicitly set - all features are available
```

Comment:
This is potential source of issues. I'd rather avoid to allow pass `null` here. Might be good idea to ensure the parameters passed here aren't null

Reply:
Only ` public IndexWriter([NotNull] Transaction tx, IndexFieldsMapping fieldsMapping, in SupportedFeatures supportedFeatures = default) : this(fieldsMapping, supportedFeatures)` is used by Raven, and it's only 1 call, rest calls/ctors are used by tests.

Reply:
so can we add `new SupportedFeatures()`  in tests?

Reply:
1) `SupportedFeatures` is now mandatory for all ctors
2) Introduced static readonly `SupportedFeatures.All` field and used it in all tests.